iop/memorycard/mcman/src/mcsio2.c

```diff
@@ -519,14 +530,19 @@ int mcman_eraseblock(int port, int slot, int block, void **pagebuf, void *eccbuf
 		if (((mcman_sio2packet.stat6c & 0xF000) != 0x1000))
 			continue;
 
-		if (p[3] == 0x5a)
+		if (p[3] == 0x5a) {
+            HAKAMA_SIGNALSEMA();
 			return sceMcResSucceed;
+        }
 	} while (++retries < 100);
 
-	if (p[3] == 0x66)
+    HAKAMA_SIGNALSEMA(); //El_isra: Original impl has two semas (still commented there) replaced with one for simplicity
```

Comment:
I think you can't signal before as it could potentially change the value of `p` before you read in the following "ifs"